compiler/rustc_lint/src/traits.rs

```diff
@@ -132,3 +134,44 @@ impl<'tcx> LateLintPass<'tcx> for DropTraitConstraints {
         }
     }
 }
+
+declare_lint! {
+    pub DUP_TRAIT_BOUNDS,
+    Deny,
+    "Duplicate trait bounds are meaningless"
+}
+
+declare_lint_pass!(
+    DuplicateTraitBounds => [DUP_TRAIT_BOUNDS]
+);
+
+impl<'tcx> LateLintPass<'tcx> for DuplicateTraitBounds {
+    fn check_item(&mut self, cx: &LateContext<'_>, item: &'tcx hir::Item<'tcx>) {
+        let bounds: &[hir::GenericBound<'_>] = match &item.kind {
+            hir::ItemKind::Trait(_, _, _, bounds, _) => bounds,
+            hir::ItemKind::TraitAlias(_, bounds) => bounds,
+            _ => return,
+        };
+
+        let mut set = FxHashSet::default();
+        for bound in bounds.into_iter() {
+            match bound {
+                GenericBound::Trait(polytraitref, _) => {
+                    let Some(did) = polytraitref.trait_ref.trait_def_id() else { continue; };
+                    // If inserting the trait bound into the set returns `false`,
+                    // there is a duplicate.
+                    if !set.insert(did) {
+                        let span = polytraitref.span;
+                        cx.struct_span_lint(DUP_TRAIT_BOUNDS, span, |lint| {
+                            let msg = format!("duplicate trait bound");
+                            lint.build(&msg)
+                                .span_help(span, "Remove this duplicate trait bound")
```

Comment:
We use lowercase throughout our messaging. It might not be great, but it's better to be consistent :)

```suggestion
                                .span_help(span, "remove the duplicate trait bound")
```